docs: Add example showing how to include sharp in custom lambda (#7522)
aws-amplify/docs: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Deploy a function with a Lambda Layer',
  description:
    'How to create a custom function resource that makes use of the Sharp library',
  platforms: [
    'android',
    'angular',
    'flutter',
    'javascript',
    'nextjs',
    'react',
    'react-native',
    'swift',
    'vue'
  ]
};

export function getStaticPaths() {
  return getCustomStaticPath(meta.platforms);
}

export function getStaticProps(context) {
  return {
    props: {
      meta
    }
  };
}

Amplify Functions are powered by [AWS Lambda](https://aws.amazon.com/lambda/), and allow you to perform a wide variety of customization through self-contained _functions_. However, there are times when a feature may not be directly supported through what Amplify exposes. Fortunately, AWS Amplify allows the use of [custom resources](https://docs.amplify.aws/react/build-a-backend/add-aws-services/custom-resources/) to be created. This enables the full flexibility of the [AWS CDK](https://docs.aws.amazon.com/cdk/api/v2/), while still providing a type-safe environment.

> In this section, we'll explore this feature by creating a custom Lambda function that bundles the [Sharp nodejs package](https://www.npmjs.com/package/sharp).

Reviewer: cut from here and put it right below the next heading

To get started, ensure you have a Sharp layer deployed to your AWS account.

> ## Sharp Layer Deployment

Reviewer: Step 1 - Deploy your Lambda layer

If you do not have a Sharp layer deployed in your AWS account, you can deploy it using the following steps:

1. Visit the [AWS Sharp Layer Repository](https://github.com/Umkus/lambda-layer-sharp).
2. In the README, click on the "releases" link and download the latest `sharp` zip file to your computer.
3. Open your terminal and execute the following command to deploy the Sharp package to AWS Lambda:

> Feel free to adjust the location of the `zip-file` as needed. Also, if using the `AmplifySet` role, you may need to adjust your permissions.

```sh title="In your terminal"
aws lambda publish-layer-version \
    --layer-name sharp \
    --description "Sharp layer" \
    --license-info "Apache License 2.0" \
    --zip-file fileb://dist/sharp-layer.zip \
    --compatible-runtimes nodejs18.x nodejs20.x \
    --compatible-architectures x86_64 arm64
```

Once deployed, you'll see the ARN of your Lambda layer similar to below:

> ```sh title="example lambda layer arn"

Reviewer: ```suggestion
```console title="Terminal"
```

arn:aws:lambda:us-east-1:211125654548:layer:sharp
```

> It's *very* important to note that we'll needs to specify the version in addition to this ARN. In this example, we'll be adding this layer to a `.env.local` file with the version appended as shown:

Reviewer: ```suggestion
It's **very** important to specify the version in addition to this ARN. In this example, we'll be adding this layer to a `.env.local` file with the version appended as shown:
```

```sh title=".env.local"
SHARP_LAYER_ARN=arn:aws:lambda:us-east-1:211125654548:layer:sharp:1
```

> ## Creating a custom Lambda Function with a Lambda Layer

Reviewer: ```suggestion
## Step 2 - Create a custom Lambda function with a Lambda layer
```

To begin, we'll create a separate stack to contain our function:

```ts title="amplify/backend.ts"
const backend = defineBackend({
	storage, //example resource
})

const customResourceStack = backend.createStack(
	"CustomResources"
)
```

Now, we'll simply add our layer to our custom stack, in addition to the Lambda function:

```ts title="amplify/backend.ts"
import {
	LayerVersion,
	Runtime,
} from 'aws-cdk-lib/aws-lambda'
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs'
import * as path from 'path'

const __dirname = import.meta.dirname

//...rest of code

const sharpLayer = LayerVersion.fromLayerVersionArn(
  customResourceStack,
  `myapp-sharpLayer`,
  process.env.SHARP_LAYER_ARN!
)

const myCustomNodeJSFunctionWithLayer = new NodejsFunction(
  scope,
  'my-custom-nodejs-function-with-layer',
  {
    functionName: 'my-custom-nodejs-function-with-layer',
    entry: path.join(__dirname, 'function-main.ts'),
    runtime: Runtime.NODEJS_20_X,
    layers: [sharpLayer],
    bundling: {
      externalModules: ['sharp'],
    },
  }
)
```

> Above you can see that we are giving our function a name, where it's handler function is located, the runtime, and that our sharp layer is to be used and considered as externally brought in as a dependency.

Reviewer: ```suggestion
Above you can see that we are giving our function a name – where its handler function is located – the runtime, and that our sharp layer is to be used and considered as externally brought in as a dependency.
```

Locally, you will likely want to test out the Sharp library. This can be done by simply installing it as a dev dependency:

> ```sh title="installing sharp"

Reviewer: ```suggestion
```sh title="Terminal"
```
